restructure CLAUDE.md (#382)
posit-dev/ggsql: ggsql-wasm/CLAUDE.md

## Build

The full build:

```sh
cd ggsql-wasm
./build-wasm.sh
```

This sequentially:

1. `npm install && npm run build` in `library/` — produces the typed JS wrapper.
2. `wasm-pack build --target web --profile wasm --no-opt` — compiles `src/lib.rs` to `pkg/`. The `wasm` profile is defined in the workspace `Cargo.toml` (release-style, `opt-level = "z"`, LTO, `panic = "abort"`).
3. `wasm-opt pkg/ggsql_wasm_bg.wasm -o pkg/ggsql_wasm_bg.wasm -Oz` — shrinks the binary further.
4. `npm install && npm run build` in `demo/` — bundles the playground UI.
5. Copies `demo/dist/` to `/doc/wasm/` so Quarto can serve it under the docs site.

Flags:

- `--skip-binary` — reuse the existing `pkg/` (skip steps 2–3); useful when iterating on `library/` or `demo/`.
- `--skip-opt` — compile but skip `wasm-opt` (faster, larger binary).

## Wasm-specific feature constraints

`Cargo.toml` carves out wasm32-only dependency overrides:

- `getrandom` and `uuid` are forced to the `js` feature so they get randomness from the browser.
- `sqlite-wasm-rs` replaces `rusqlite` for SQLite support in the browser.
- `tokio` is reduced to `default-features = false` (no I/O reactor on wasm).

> DuckDB and ODBC are not enabled here — they require host APIs that aren't available in the browser.

Reviewer: ```suggestion
ODBC is not enabled here — it requires host APIs that aren't available in the browser.
```

We'll probably add duckdb back at some point.

Author: but aren't right now, right? I think it should reflect current status
